src/cobra/flux_analysis/moma.py

```diff
@@ -114,7 +114,7 @@ def add_moma(model, solution=None, linear=True):
         raise ValueError("model is already adjusted for MOMA")
 
     # Fall back to default QP solver if current one has no QP capability
-    if not linear:
+    if not linear and sutil.interface_to_str(model.problem) not in sutil.qp_solvers:
```

Comment:
I know that's not part of your PR here but I was wondering if it'd better behaviour to raise an exception rather than automatically switching solver backend. What do you think?

Reply:
No, that is a good point. There is also a bit of a new issue with the new solvers like OSQP and CBC. Those are specialized solvers that are good for one type of problem but don't support the full set of cobrapy operations:

- OSQP is great for QPs but does not get accurate <1e-3 for most LP problems
- CBC is great for MIP problems but does not allow renaming variables

So those are best used for some kind of analysis and changed afterward, but I'm not sure how to handle that. Maybe a warning/info on switching to them?

Reply:
I think that a warning is a good option. Both when switching a model's solver, as well as when setting them as the default in the `Configuration`.

Reply:
Awesome, I'll do that then!